assets/app/scripts/controllers/attachPersistentVolumeClaim.js

```diff
@@ -0,0 +1,280 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateRouteController
+ * @description
+ * # CreateRouteController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('AttachPersistentVolumeClaimController', function ($filter, $routeParams, $scope, $window, StorageService, DataService, Navigate, ProjectsService) {
+    $scope.alerts = {};
+    $scope.renderOptions = {
+      hideFilterWidget: true
+    };
+
+    $scope.projectName = $routeParams.project;
+    $scope.deploymentConfigName = $routeParams.deploymentconfig;
+    $scope.deploymentName = $routeParams.deployment;
+
+    $scope.attach = {
+      deploymentConfig: null,
+      deployment: null,
+      persistentVolumeClaim: null,
+      volumeName: null,
+      mountPath: null,
+      containers: {
+        all: true,
+        individual: {}
+      }
+    };
+
+    $scope.breadcrumbs = [
+      {
+        title: $routeParams.project,
+        link: "project/" + $routeParams.project
+      },
+      {
+        title: "Attach Storage"
+      }
+    ];
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+
+        var containerToAttachProvided = function() {
+          for (var key in $scope.attach.containers.individual) {
+            if ($scope.attach.containers.individual[key] === true) {
+              return true;
+            }
+          }
+          return false;
+        };
+
+        var sendError = function(errorMessage, errorDetails) {
+          $scope.disableInputs = true;
+          $scope.alerts['attach-persistent-volume-claim'] = {
+            type: "error",
+            message: errorMessage,
+            details: errorDetails
+          };
+        };
+
+        // check parameters validity
+        var validate = function() {
+          if ($scope.deploymentConfigName && $scope.deploymentName) {
+            sendError("Deployment and deployment config can't be both provided provided.");
+            return false;
+          }
+          if (!$scope.deploymentConfigName && !$scope.deploymentName) {
+            sendError("A deployment or deployment config must be provided.");
+            return false;
+          }
```

Comment:
> These errors only occur when the query parameters are incorrect, right?

If true, I'd change the name to something more specific like `validateRouteParams` to avoid confusion with form validation and perhaps check the route params directly at the top of the file before assigning scope values.